src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -501,6 +499,41 @@ function reducer(draft, action) {
             draft.logicRulesToDelete = [];
             break;
         }
+        /**
+         * Form Variables
+         */
+        case 'ADD_USER_DEFINED_VARIABLE': {
+            let updatedVariables = _.cloneDeep(draft.formVariables);
```

Comment:
why do we need the `cloneDeep`? immer-reducer is supposed to watch for changes to the internal datastructures, so you should be able to do `draft.formVariables.push(EMPTY_VARIABLE)`